### DISCUSSION TITLE
Multiple DialogPanel elements not supported

### DISCUSSION THREAD

**Discussion by JokerMartini** [tailwindlabs/headlessui#2724] [Ideas]
It appears that when you add more than a single DialogPanel within a Dialog it does not take into account the click events on all the DialogPanels. It closes the Dialog if you click anywhere except on the last created DialogPanel within the component. It would be nice if you could support multiple DialogPanel's within a single Dialog.

**Comment by adamwathan** [comment]
Hey! Can you share your use case? I never in my life could have predicted someone would want this 😅

**Reply by JokerMartini** [reply]
I understand, i was in the same boat until I tried to recreate a LightBox type of plugin using the Dialog. Mimicking what you see on populate websites. Here is my problem. I need a button panel on the **left panel**, **right panel** and the a **center content** section which contains the image. I wanted the negative space to still allow for users to click and close. Here is an image showcasing the areas marked in green which i wanted to be created using multiple Dialog Panels. 

In one case i wanted to add a floating button in the top right that users could click, bit that didn't work either it also needed to be constructed using two Dialog Panels.

![image](https://github.com/tailwindlabs/headlessui/assets/1416749/60d9b108-e617-4ed5-8148-c672df71ef6d)

Here is the other scenario i was talking about...

![image](https://github.com/tailwindlabs/headlessui/assets/1416749/a93ff7ce-1406-46ee-915d-7c7393215645)
